src/pages/en/guides/migration-guide/from-gatsby.md

```diff
@@ -0,0 +1,1187 @@
+
+#### Migrating Custom a Custom 404 Page 
+
+To [create a custom 404 page with Astro](https://docs.astro.build/en/core-concepts/astro-pages/#custom-404-error-page), create a file called `src/pages/404.astro`. Most [deploy services](/en/guides/deploy/) will find and use it.
+
+### Migrating Tests
+
+As Astro outputs raw HTML, it's possible to write end-to-end tests utilizing the output of the build step. Any end-to-end tests written previously should, potentially, work out-of-the-box, assuming you've been able to match the markup of the older Gatsby site.
+
+See our [testing guide](https://docs.astro.build/en/guides/testing/) for more.
+
+## Convert Syntax to Astro
+
+### Gatsby Links to Astro
+
+Convert any Gatsby `<Link to="">`, `<NavLink>` etc. components to HTML `<a href="">` tags. 
+
+Astro does not use any special component for links, although you are welcome to build custom link components. You can then import and use this `<Link>` just as you would any other component.
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Gatsby Imports to Astro
+
+Astro requires file imports to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full (e.g. `../../layouts/Layout.astro`). Note that `.astro` and several other file types must be imported with their full file extension.
+
+### Gatsby Children to Astro
+
+Convert any instances of `{children}` to an Astro `<slot />`. Astro does not need to receive `{children}` as a function prop and will automatically render child content in a `<slot />`.
+
+React components that pass multiple sets of children can be migrated to an Astro component using [named slots](https://docs.astro.build/en/core-concepts/astro-components/#named-slots):
+
+```jsx title="pageheader.jsx"
+export const PageHeader = ({navItems, children}) => {
+	return (
+		<header>
+			{children}
+			<nav>
+				<ul>
+					<li><Link to="/">Home</a></li>
+					{navItems}
+				</ul>
+			</nav>
+		</header>
+	)
+}
+```
+
+```jsx title="app.jsx"
+import {PageHeader} from './pageheader';
+
+export const App = () => {
+	return (
+		// ...
+		<PageHeader navItems={<li><Link to="/blog">Blog</a></li>}>
+			<Link to="/">CompanyCo</a>
+		</PageHeader>
+		// ...
+	)
+}
+```
+
+
+
+This can be migrated to an Astro component using [named slots](https://docs.astro.build/en/core-concepts/astro-components/#named-slots):
+
+```astro title="src/components/PageHeader.astro"
+<header>
+    <nav>
+        <ul>
+            <li><a href="/">Home</a></li>
+            <slot name="navItems"/>
+        </ul>
+    </nav>
+	<slot/>
+</header>
+```
+
+```astro title="src/components/App.astro"
+---
+import PageHeader from './PageHeader.astro';
+---
+
+<PageHeader>
+	<a href="/">CompanyCo</a>
+	<li slot="navItems"><a href="/blog">Blog</a></li>
+</PageHeader>
+
+<!-- ... -->
+```
```

Comment:
``````suggestion
The above component can be reproduced in Astro by using a `slot=""` attribute to pass data to a particular named `<slot />` placeholder component.

```astro title="src/components/PageHeader.astro"
<header>
    <nav>
        <ul>
            <li><a href="/">Home</a></li>
            <slot name="navItems"/>
        </ul>
    </nav>
	<slot/>
</header>
```

```astro title="src/components/Component.astro"
---
import PageHeader from './PageHeader.astro';
---

<PageHeader>
	<a href="/">CompanyCo</a>
	<li slot="navItems"><a href="/blog">Blog</a></li>
</PageHeader>

<!-- ... -->
```
See more about [specific `<slot />` usage in Astro](/en/core-concepts/astro-components/#slots) for details.
``````
Just me or maybe let's not do `App.astro`??  😅